src/common/transformations/src/transformations/low_precision/mark_dequantization_subgraph.cpp

```diff
@@ -10,9 +10,24 @@
 #include <transformations/rt_info/dequantization_node.hpp>
 #include <transformations/rt_info/disable_constant_folding.hpp>
 
-#include "bound_evaluation_util.hpp"
+static bool is_constfoldable(const std::shared_ptr<ov::Node>& node) {
+    if (ov::is_type<ov::opset10::Constant>(node))
+        return true;
 
-using namespace ngraph;
+    const auto num_inputs = node->get_input_size();
+    if (num_inputs == 0)
+        return false;
+
+    if (!node->has_evaluate() || ov::constant_folding_is_disabled(node))
+        return false;
+
+    for (size_t i = 0; i < num_inputs; i++) {
+        const auto& input = node->get_input_node_shared_ptr(i);
+        if (!is_constfoldable(input))
+            return false;
+    }
+    return true;
+}
```

Comment:
Can you, please, copy exists and well tested not recursive `ngraph::could_propagate` function and just remove `ShapeOf` logic, please?

Reply:
done